### DISCUSSION TITLE
[Deployment] deploy L2Erc20bridge through zksync-era

### DISCUSSION THREAD

**Discussion by Hedy-chu** [zkSync-Community-Hub/zksync-developers#624] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [X] Smart contracts (era-contracts)
- [X] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.75.0

### Operating System

- [ ] Ubuntu
- [X] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

i want deploy the official L2erc20bridge of zksync via zksync-era in zksyncTestnet，and i find need use `zk` command，but i don‘t know how can i use `zk` commond ，does i need installing all dependencies like this：https://github.com/matter-labs/zksync-era/blob/main/docs/guides/setup-dev.md

### Expected Behavior

 deploy L2erc20bridge and other zksync office smart contracts

### Current Behavior

i have pull zksync-era repo and submoudle submoudle，but i don't know the next step

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment, marked as answer]
To deploy the official L2Erc20bridge and other zkSync official smart contracts on the zkSync Testnet using the `zk` command, you'll need to follow a series of steps to set up your environment and execute the deployment. Here’s a step-by-step guide to help you through the process:

### Step 1: Install Dependencies
Before you can use the `zk` command, ensure that all necessary dependencies are installed. Since you are using MacOS, follow the setup guide specific to your operating system. You can find the detailed instructions on setting up your development environment in the zkSync Era documentation [here](https://github.com/matter-labs/zksync-era/blob/main/docs/guides/setup-dev.md).

### Step 2: Clone the Repository
You mentioned that you have already pulled the zkSync Era repository and its submodules. If anyone else needs this step, they can clone the repository with:
```bash
git clone --recurse-submodules https://github.com/matter-labs/zksync-era.git
cd zksync-era
```

### Step 3: Using the `zk` Command
The `zk` command is a part of the zkSync tooling. To use it, you might need to build or install the specific tools from the zkSync Era repository. Typically, this involves running build commands in the repository directory. Look for a `Makefile` or similar instructions in the repository to compile or set up the `zk` tool.

### Step 4: Deploy the Smart Contracts
Once the `zk` tool is set up, you can use it to deploy smart contracts. The exact command will depend on the specifics of the tooling provided in the zkSync Era repository. Generally, you would use a command similar to:
```bash
zk deploy --contract L2Erc20Bridge --network zksyncTestnet
```
This is a generic example. You will need to refer to the specific documentation or help command (`zk --help`) provided by the zkSync Era tooling for exact syntax and options.

### Step 5: Verify the Deployment
After deploying the contracts, verify that they are correctly deployed on the zkSync Testnet. You can use the zkSync Era block explorer or similar tools to check the contract address and ensure that it is interacting as expected.

### Additional Resources
- **zkSync Era Documentation**: Make sure to frequently consult the [zkSync Era documentation](https://docs.zksync.io/) for any updates or changes to the deployment processes.
- **Community and Support**: If you encounter specific issues or errors during the deployment, consider reaching out on the official [zkSync community forums](https://github.com/zkSync-Community-Hub/zksync-developers/discussions) or filing an issue in the repository for more targeted help.

By following these steps, you should be able to deploy the L2Erc20bridge and other official smart contracts on the zkSync Testnet.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.
